tests/unit/components/NcAvatar/NcAvatar.spec.ts

```diff
@@ -20,11 +20,102 @@
  *
  */
 
-import { mount } from '@vue/test-utils'
+import { mount, shallowMount } from '@vue/test-utils'
 import { nextTick } from 'vue'
 import NcAvatar from '../../../../src/components/NcAvatar/NcAvatar.vue'
 
+interface ITestCase {
+	name: string
+	expected: string
+	props: Record<string, unknown>
+}
+
 describe('NcAvatar.vue', () => {
+	describe('Initials', () => {
+		const testCases: ITestCase[] = [
+			{
+				name: 'empty user',
+				expected: '?',
+				props: {},
+			},
+			{
+				name: 'user property',
+				expected: 'JD',
+				props: {
+					user: 'Jane Doe',
+					isNoUser: true,
+				},
+			},
+			{
+				name: 'display name property',
+				expected: 'JD',
+				props: {
+					displayName: 'Jane Doe',
+				},
+			},
+			{
+				name: 'display name property over user property',
+				expected: 'NU',
+				props: {
+					displayName: 'No User',
+					user: 'I am a group',
+					isNoUser: true,
+				},
+			},
+			{
+				name: 'special characters in name',
+				expected: 'JD',
+				props: {
+					displayName: 'Jane (Doe)',
+				},
+			},
+			{
+				name: 'lower case name',
+				expected: 'JD',
+				props: {
+					displayName: 'jane doe',
+				},
+			},
+			{
+				name: 'middle names',
+				expected: 'JD',
+				props: {
+					displayName: 'Jane Some Name Doe',
+				},
+			},
+			{
+				name: 'non ascii characters',
+				expected: 'ÜÖ',
+				props: {
+					displayName: 'Ümit Öçal',
+				},
+			},
+			{
+				name: 'non latin characters',
+				expected: 'ジド',
+				props: {
+					displayName: 'ジェーン ドー',
+				},
+			},
+		]
+
+		for (const { name, props, expected } of testCases) {
+			it(`can handle ${name}`, async () => {
+				const wrapper = shallowMount(NcAvatar, {
+					propsData: {
+						...props,
+					},
+				})
+				expect(wrapper.exists()).toBe(true)
+
+				await nextTick()
+				const initials = wrapper.find('.avatardiv__initials')
+				expect(initials.exists()).toBe(true)
+				expect(initials.text()).toBe(expected)
```

Comment:
IMO, it is not good to test span by class name `.avatardiv__initials` here, it is not a component interface. As all the avatar is a block with initials, it seems, we can just check the text in the avatar.

And is there a case where a wrapper could not exist in VTU?

```suggestion
				await nextTick()
				expect(wrapper.text()).toBe(expected)
```

Reply:
I agree better are test selectors like `data-testid` but in this case nothing is needed :)